mypy/checker.py

```diff
@@ -1676,6 +1677,60 @@ def visit_class_def(self, defn: ClassDef) -> None:
         if typ.is_protocol and typ.defn.type_vars:
             self.check_protocol_variance(defn)
 
+    def check_init_subclass(self, defn: ClassDef) -> None:
+        """
+        Check that keywords in a class definition are valid arguments for __init_subclass__().
+
+        In this example:
+            1   class Base:
+            2       def __init_subclass__(cls, thing: int):
+            3           pass
+            4   class Child(Base, thing=5):
+            5       def __init_subclass__(cls):
+            6           pass
+            7   Child()
+
+        Base.__init_subclass__(thing=5) is called at line 4. This is what we simulate here.
+        Child.__init_subclass__ is never called.
+        """
+        typ = defn.info
+        # At runtime, only Base.__init_subclass__ will be called
+        # we skip the current class itself.
+        found = False
+        for base in typ.mro[1:]:
+            # there are "NOT_READY" instances
+            # during the tests, so I filter them out...
+            if base.defn.info:
```

Comment:
I think I have an idea why this may be missing, but anyway this is probably not important since you don't need this anyway, everywhere where you use `base.defn.info` you can use just `base`, this is the same object.

Reply:
understood!